modules/system/classes/CombineAssets.php

```diff
@@ -338,19 +338,32 @@ protected function prepareAssets(array $assets)
         /*
          * Determine which group of assets to combine.
          */
-        if (count($combineCss) > count($combineJs)) {
+        $cssCount = count($combineCss);
+        $jsCount = count($combineJs);
+        if ($cssCount > $jsCount) {
             $extension = 'css';
             $assets = $combineCss;
         }
-        else {
+        elseif ($cssCount < $jsCount) {
             $extension = 'js';
             $assets = $combineJs;
         }
+        else {
+            $extension = null;
+            $assets = array_unique([...$combineCss, ...$combineJs]);
+        }
 
         /*
          * Apply registered aliases
          */
         if ($aliasMap = $this->getAliases($extension)) {
+            if ($extension === null) {
+                $aliasMap = array_map(function ($map) {
+                    return is_array($map) ? $map : [$map];
+                }, $aliasMap);
+                $aliasMap = array_reduce($aliasMap, 'array_merge', []);
+            }
```

Comment:
What does this section do?

Reply:
When using [the `getAliases` method](https://github.com/wintercms/winter/blob/6e8c8d90f11cf38560f3e5f1f1fe3ffac904086c/modules/system/classes/CombineAssets.php#L765-L776) (to build the `$aliasMap` on the line 361) with a null value, it returns the whole aliases array with extensions as keys:
```php
'css' => [
  'framework.extras' => '~/modules/system/assets/css/framework.extras.css',
  'framework.extras.css' => '~/modules/system/assets/css/framework.extras.css',
  'snowboard.extras.css' => '~/modules/system/assets/css/snowboard.extras.css'
],
'js' => [
  'framework.extras' => '~/modules/system/assets/js/framework.extras.js',
  'framework.extras.js' => '~/modules/system/assets/js/framework.extras.js',
  'snowboard.extras.js' => '~/modules/system/assets/js/snowboard.extras.js'
]
// other extensions aliases ...
```
I don't know at this level which extensions are registered, so I used the `array_reduce` combined with the `array_merge` to not get any responsibility for this.
This results to the one-level array in `$aliasMap` that we need:
```php
[
  'framework.extras.css' => '~/modules/system/assets/css/framework.extras.css',
  'snowboard.extras.css' => '~/modules/system/assets/css/snowboard.extras.css',
  'framework.extras' => '~/modules/system/assets/js/framework.extras.js',
  'framework.extras.js' => '~/modules/system/assets/js/framework.extras.js',
  'snowboard.extras.js' => '~/modules/system/assets/js/snowboard.extras.js'
]
```
_This is the reason of the `framework.extras` js alias dominance over the css one.`_

The reason why I needed to use the "array casting" on line 361-363 into the `array_map` is that I faced something that wasn't an array, but re-reading at the code of `registerAlias`, I don't get how this could happen. I will take a deeper look to understand what happened.